pkg/plugins/runtime/k8s/probes/probe_converter_test.go

```diff
@@ -0,0 +1,191 @@
+package probes_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	kube_core "k8s.io/api/core/v1"
+	"sigs.k8s.io/yaml"
+
+	"github.com/kumahq/kuma/pkg/plugins/runtime/k8s/probes"
+)
+
+var _ = Describe("KumaProbe", func() {
+	Describe("OverridingSupported", func() {
+		type testCase struct {
+			input             kube_core.Probe
+			expectedSupported bool
+		}
+
+		DescribeTable("should check if probe is supported to be overridden",
+			func(given testCase) {
+				virtual := probes.KumaProbe(given.input)
+
+				Expect(virtual.OverridingSupported()).To(Equal(given.expectedSupported))
+			},
+			Entry("HTTP", testCase{
+				input: kube_core.Probe{
+					ProbeHandler: kube_core.ProbeHandler{
+						HTTPGet: &kube_core.HTTPGetAction{},
+					},
+				},
+				expectedSupported: true,
+			}),
+			Entry("TCPSocket", testCase{
+				input: kube_core.Probe{
+					ProbeHandler: kube_core.ProbeHandler{
+						TCPSocket: &kube_core.TCPSocketAction{},
+					},
+				},
+				expectedSupported: true,
+			}),
+			Entry("gRPC", testCase{
+				input: kube_core.Probe{
+					ProbeHandler: kube_core.ProbeHandler{
+						GRPC: &kube_core.GRPCAction{},
+					},
+				},
+				expectedSupported: true,
+			}),
+			Entry("exec", testCase{
+				input: kube_core.Probe{
+					ProbeHandler: kube_core.ProbeHandler{
+						Exec: &kube_core.ExecAction{},
+					},
+				},
+				expectedSupported: false,
+			}))
+	})
+
+	Context("ToVirtual - HTTP", func() {
+		It("should convert pod probe to virtual probe", func() {
+			podProbeYaml := `
+                httpGet:
+                  path: /c1/health/liveness
+                  port: 8080
+`
+			probe := kube_core.Probe{}
+			err := yaml.Unmarshal([]byte(podProbeYaml), &probe)
+			Expect(err).ToNot(HaveOccurred())
+
+			virtual, err := probes.KumaProbe(probe).ToVirtual(9000)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(virtual.Path()).To(Equal("/8080/c1/health/liveness"))
+			Expect(virtual.Port()).To(Equal(uint32(9000)))
+		})
+
+		It("should convert probe with custom headers and timeout", func() {
+			podProbeYaml := `
+                timeoutSeconds: 15
+                httpGet:
+                  scheme: HTTPS
+                  path: /c1/healthz
+                  port: 8080
+                  httpHeaders:
+                  - name: Host
+                    value: example.com
+                  - name: X-Custom-Header
```

Comment:
We don't copy custom headers?

Reply:
We do. I forgot to assert it. I'll add it.
Thanks.